### DISCUSSION TITLE
Save when database is locked

### DISCUSSION THREAD

**Discussion by LewisOnGit** [Kunzisoft/KeePassDX#1255] [Ideas]
Hi! I use KeepassXC on my computer and it has an option in the settings which allows you to make edits to your database without waiting for it to save after each change, then when the database is locked/closed it automatically saves all your changes.

I'm finding KeepassDX quite slow on my phone because I have to wait for the database to save after every change I make and I don't want to rely on myself to manually save the database before closing it.

**Comment by J-Jamet** [comment]
Indeed, this is a feature to study.
I have vaguely thought about it and we have to adapt the architecture so that the broadcast that causes the lock event goes through the service and the backup thread before doing the real lock.

**Reply by LewisOnGit** [reply]
Could it be adapted from the KeepassXC source code?

**Comment by nextturnp** [comment]
On ၂၀၂၂၊ ၃ မတ် GMT+06:30 ၀၄:၄၆:၅၇, Lewis Michael Harold ***@***.***> wrote:
>Could it be adapted from the KeepassXC source code?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/Kunzisoft/KeePassDX/discussions/1255#discussioncomment-2283730
>You are receiving this because you are subscribed to this thread.
>
>Message ID:
***@***.***>

--  Sent from my p≡p for Android.

**Reply by LewisOnGit** [reply]
?
